test/cluster/functionality.sh

```diff
@@ -0,0 +1,114 @@
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[cluster\.service\s*\]"
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[node\s*\]\s*\[.*\]\s*started"
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} recovered its indices after starting..."
+	if oc logs "${elasticsearch_pod}" | grep -q "detected_master"; then
+		# If this instance detects a different master,
+		# it won't recover its own indices
+		os::cmd::expect_success_and_text "oc logs ${elasticsearch_pod}" "\[gateways*\]\s*\[.*\]\s*recovered\s*\[\d*\]\s*indices into cluster_state"
+	elif oc logs "${elasticsearch_pod}" | grep -q "new_master"; then
+		# If we aren't master we should be started by
+		# now and should have detected a master
+		os::log::fatal "ElasticSearch pod ${elasticsearch_pod} isn't master and was unable to detect master"
+	fi
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} contains common data model index templates..."
+	os::cmd::expect_success "oc exec ${elasticsearch_pod} -- ls -1 /usr/share/elasticsearch/index_templates"
+	for template in $( oc exec "${elasticsearch_pod}" -- ls -1 /usr/share/elasticsearch/index_templates ); do
+		os::cmd::expect_success_and_text "oc exec ${elasticsearch_pod} -- curl -sk --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key -X HEAD -w '%{response_code}' https://localhost:9200/_template/${template}" '200'
+	done
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} has persisted indices created by Fluentd..."
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "update_mapping \[com\.redhat\.viaq\.common\]"
+	# We are interested in indices with one of the following formats:
+	#     .operations.<year>.<month>.<day>
+	#     project.<namespace>.<uuid>.<year>.<month>.<day>
+	for index in $( oc exec "${elasticsearch_pod}" -- curl -sk --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key https://localhost:9200/_cat/indices?h=index ); do
+		if [[ "${index}" == ".operations"* ]]; then
+			# If this is an operations index, we will be searching
+			# on disk for it
+			index_search_path="/var/log/messages"
+		elif [[ "${index}" == "project."* ]]; then
+			# Otherwise, we will find it in the container log, which
+			# we can identify with the UUID
+			uuid="$( cut -d '.' -f 3 <<<"${index}" )"
+			index_search_path="/var/log/containers/*_${uuid}_*.log"
+		else
+			continue
+		fi
+
+		# We don't care about the date in the index
+		index="$( rev <<<"${index}" | cut -d"." -f 4- | rev )"
+
+		for kibana_pod in $( oc get pods --selector component="${OAL_KIBANA_COMPONENT}"  -o jsonpath='{ .items[*].metadata.name }' ); do
+			os::log::info "Cheking for index ${index} with Kibana pod ${kibana_pod}..."
+			# As we're checking system log files, we need to use `sudo`
+			sudo -E VERBOSE=true go run "${OS_O_A_L_DIR}/hack/testing/check-logs.go" "${kibana_pod}"        \
```

Comment:
Is this a hard failure?  I have found this code fraught with peril - it is very difficult to do accurate string comparisons like this.